api/resource.rb

```diff
@@ -26,52 +26,94 @@ class Resource < Api::Object::Named
+      attr_reader :references
+      # [Required] The GCP "relative URI" of a resource, relative to the product
+      # base URL. It can often be inferred from the `create` path.
       attr_reader :base_url
-      # URL to use for creating the resource. If not specified, the
-      # collection url (when create_verb is default or :POST) or
-      # self_link (when create_verb is :PUT) is used instead.
+
+      # ====================
+      # Common Configuration
+      # ====================
+      #
+      # [Optional] The minimum API version this resource is in. Defaults to ga.
+      attr_reader :min_version
+      # [Optional] If set to true, don't generate the resource.
+      attr_reader :exclude
+      # [Optional] If set to true, the resource is not able to be updated.
+      attr_reader :input
+      # [Optional] If set to true, this resource uses an update mask to perform updates.
+      # This is typical of newer GCP APIs.
+      attr_reader :update_mask
+      # [Optional] If set to true, the object has a `self_link` field. This is
+      # typical of older GCP APIs.
+      attr_reader :has_self_link
+
+      # ====================
+      # URL / HTTP Configuration
+      # ====================
+      #
+      # [Optional] The "identity" URL of the resource.. Defaults to:
+      # * collection url when the create_verb is :POST
+      # * self_link when the create_verb is :PUT
+      attr_reader :self_link
+      # [Optional] The URL used to creating the resource. Defaults to:
+      # * collection url when the create_verb is :POST
+      # * self_link when the create_verb is :PUT
       attr_reader :create_url
-      # URL to use to delete the resource. If not specified, the
-      # self link is used.
+      # [Optional] The URL used to delete the resource. Defaults to the self
+      # link.
       attr_reader :delete_url
-      # URL to use for updating the resource. If not specified, the self link
-      # will be used. This currently can only be used with Terraform resources.
-      # TODO(#302): Add support for the other providers.
+      # [Optional] The URL used to update the resource. Defaults to the self
+      # link.
       attr_reader :update_url
-      attr_reader :self_link
-      # This is the type of response from the collection URL. It contains
-      # the name of the list of items within the json, as well as the
-      # type that this list should be. This is of type Api::Resource::ResponseList
+      # [Optional] The HTTP verb used during create. Defaults to :POST.
+      attr_reader :create_verb
+      # [Optional] The HTTP verb used during update. Defaults to :PUT.
+      attr_reader :update_verb
+      # [Optional] The HTTP verb used during delete. Defaults to :DELETE.
+      attr_reader :delete_verb
+
+      # ====================
+      # Collection / Identity URL Configuration
+      # ====================
+      #
+      # [Optional] This is the type of response from the collection URL. It
```

Comment:
```suggestion
      # [Optional] (Api::Resource::ResponseList) This is the type of response from the collection URL. It
```
What do you think about this? The type information would be easy to find + in a standardized place.

Reply:
Done